dev.c

```diff
@@ -164,24 +164,31 @@ static void fuse_put_unique(struct fuse_conn *fc, u64 uid)
 	put_cpu();
 }
 
+static struct rdwr_in *fuse_rdwr(struct fuse_conn *fc, uint32_t read)
+{
+	return &fc->queue->requests[read & (FUSE_REQUEST_QUEUE_SIZE - 1)];
+}
+
 static void queue_request(struct fuse_conn *fc, struct fuse_req *req)
 {
-	u32 write, next_index;
-
-	spin_lock(&fc->queue->w.lock);
-	write = fc->queue->w.write;
-	next_index = (write + 1) & (FUSE_REQUEST_QUEUE_SIZE - 1);
-	if (fc->queue->w.read == next_index) {
-		fc->queue->w.read = fc->queue->r.read;
```

Comment:
whats the original logic here? when a request gets queued from within the kernel, if the next index matches w.read, I assumed the queue is full. But l do not understand the next piece of logic to correlate with r.read. can you please elaborate.